lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -262,3 +262,38 @@ def _get_enrollments(self, user=None, external_user_key=None):
         ).prefetch_related('program_course_enrollments')
         serialized = ProgramEnrollmentSerializer(program_enrollments, many=True)
         return serialized.data
+
+
+class SAMLProvidersWithOrg(APIView):
+    """
+    Support-only API View for fetching a list of all
+    organizations names which will be utilized as keys.
+    """
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        """
+        The get request returns a list of all
+        organizations names which will be utilized as keys.
+        * Example Request:
+            - GET /support/get_saml_providers/
+        * Example Response:
+            [
+                'test_org',
+                'donut_org',
+                'tri_org'
+            ]
+        """
+        org_key_names = self._get_org_key_names()
+        return Response(data=org_key_names)
+
+    def _get_org_key_names(self):
+        """
+        From our Third_party_auth models, return a list of
+        of organizations names which will be utilized as keys.
+        """
+        saml_providers = SAMLProviderConfig.objects.current_set().filter(
+            enabled=True,
+            organization__isnull=False
+        ).select_related('organization')
+
+        return [saml_provider.organization.short_name for saml_provider in saml_providers]
```

Comment:
This method seems a bit redundant as its already written in `ProgramEnrollmentsInspectorView`. We can resolve this by moving this class' get method to `ProgramEnrollmentsInspectorView` and use [Viewset](https://www.django-rest-framework.org/api-guide/viewsets/) as subclass to explicitly call methods based on urls.

Reply:
Here's some suggested changes, but you can decide better. 

program_enrollments.py
```python
class ProgramEnrollmentsInspectorView(ViewSet):
    ...
    ...
    @method_decorator(require_support_permission)
    def list(self, request):
        """
        The get request returns a list of all
        organizations names which will be utilized as keys.
        * Example Request:
            - GET /support/get_saml_providers/
        * Example Response:
            [
                'test_org',
                'donut_org',
                'tri_org'
            ]
        """
        org_key_names = self._get_org_keys_and_idps().keys()
        return Response(data=org_key_names)
```

urls.py
```python
    re_path(
        r'program_enrollments_inspector/?$',
        ProgramEnrollmentsInspectorView.as_view({'get': 'get'}),
        name='program_enrollments_inspector'
    ),
    re_path(
        r'get_saml_providers/?$',
        ProgramEnrollmentsInspectorView.as_view({'get': 'list'}),
        name='get_saml_providers'
    ),
```

Reply:
`ProgramEnrollmentsInspectorView ` is a Django view as opposed to DRF view. Not certain if the above approach works but worth checking.

Reply:
So I checked this method. The method being called returns dict of `keys: objects`. For SAML, I require the list of keys only, which is a different function altogether. So, I guess I would require a separate API endpoint altogether. But yes, I can integrate it with InspectorAPIView in subsequent PR.

Reply:
I just tested above code locally and it seems to work with templates views as well. However, this can be refactored later as well.